python/ray/experimental/workflow/virtual_actor.py

```diff
@@ -0,0 +1,350 @@
+        actor_methods = inspect.getmembers(original_class,
+                                           is_function_or_method)
+        self.cls = original_class
+        self.module = original_class.__module__
+        self.name = original_class.__name__
+        self.qualname = original_class.__qualname__
+        self.methods = dict(actor_methods)
+
+        # Extract the signatures of each of the methods. This will be used
+        # to catch some errors if the methods are called with inappropriate
+        # arguments.
+        self.signatures = {}
+        for method_name, method in actor_methods:
+            # Whether or not this method requires binding of its first
+            # argument. For class and static methods, we do not want to bind
+            # the first argument, but we do for instance methods
+            method = inspect.unwrap(method)
+            is_bound = (is_class_method(method)
+                        or is_static_method(original_class, method_name))
+
+            # Print a warning message if the method signature is not
+            # supported. We don't raise an exception because if the actor
+            # inherits from a class that has a method whose signature we
+            # don't support, there may not be much the user can do about it.
+            self.signatures[method_name] = signature.extract_signature(
+                method, ignore_first=not is_bound)
+
+    def generate_random_actor_id(self) -> str:
+        """Generate random actor ID."""
+        return f"{slugify(self.qualname)}.{uuid.uuid4()}"
+
+    def flatten_args(self, method_name: str, args: Tuple[Any],
+                     kwargs: Dict[str, Any]) -> List[Any]:
+        """Check and flatten arguments of the actor method.
+
+        Args:
+            method_name: The name of the actor method in the actor class.
+            args: Positional arguments.
+            kwargs: Keywords arguments.
+
+        Returns:
+            Flattened arguments.
+        """
+        return signature.flatten_args(self.signatures[method_name], args,
+                                      kwargs)
+
+
+class VirtualActorClass:
+    """The virtual actor class used to invoke a virtual actor."""
+    _metadata: VirtualActorMetadata
+
+    def __init__(self):
+        # This shouldn't be reached because one of the base classes must be
+        # an actor class if this was meant to be subclassed.
+        assert False, (
+            "VirtualActorClass.__init__ should not be called. Please use "
+            "the @workflow.actor decorator instead.")
+
+    def __call__(self, *args, **kwargs):
+        """Prevents users from directly instantiating an ActorClass.
+
+        This will be called instead of __init__ when 'VirtualActorClass()' is
+        executed because an is an object rather than a metaobject. To properly
+        instantiated a virtual actor, use 'VirtualActorClass.create()' or
+        'VirtualActorClass.get()'.
```

Comment:
I don't think we should have a `.get` method, which is inconsistent with the existing Ray APIs of get_actor().

Why not have `workflow.get_actor` instead (or better yet, `get_or_create`).

Reply:
I'd vote to sync with ray actor API (get_actor) and maybe we can add an additional API for get_or_create ?